File: examples/python/pagination/_async/sandbox.py
import asyncio

from nightona import AsyncNightona, ListSandboxesQuery, SandboxListSortDirection, SandboxListSortField, SandboxState


async def main():
    async with AsyncNightona() as nightona:
        async for sandbox in nightona.list(
            ListSandboxesQuery(
                limit=10,
                labels={"env": "dev"},
                states=[SandboxState.STARTED],
                sort=SandboxListSortField.CREATEDAT,
                order=SandboxListSortDirection.DESC,
            )
        ):
            print(sandbox.id)


if __name__ == "__main__":
    asyncio.run(main())

File: examples/typescript/pagination/sandbox.ts
import { Nightona, SandboxListSortDirection, SandboxListSortField, SandboxState } from '@nightona-co/sdk'

async function main() {
  const nightona = new Nightona()

  for await (const sandbox of nightona.list({
    limit: 10,
    labels: { env: 'dev' },
    states: [SandboxState.STARTED],
    sort: SandboxListSortField.CREATED_AT,
    order: SandboxListSortDirection.DESC,
  })) {
    console.log(sandbox.id)
  }
}

main().catch(console.error)
